### PR TITLE
MM-46312 - Calls: Replace simple-peer with RTCPeer; use up-to-date upstream react-native-webrtc

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Use RTCPeer (as close as possible -- some changes needed: importing types from react-native-webrtc package, and being explicit about some types to make typescript happy).
- Stop using our fork of react-native-webrtc and use the up-to-date upstream release.
- Everything works after multiple tests: muting-unmuting, sharing-resharing screens.
- This lets us easily move to glare-free negotiation 🎉  (I have it in another branch, just have to setup calls/rtcd to the right versions and test).

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-46312
- https://mattermost.atlassian.net/browse/MM-42553

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 13, Pixel 6
- iOS: 15.7, iPhone 7 plus

#### Release Note

```release-note
Calls: update react-native-webrtc to most recent version
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@streamer45 @cpoile do we want this merge and shipped with today's build ?

**Comment by streamer45** [comment]
@enahum I don't think this is too urgent but would be nice to have it in for January (7.7) release. Would today be the last chance to get this in?

**Comment by enahum** [comment]
@streamer45 not the last chance no, I was just wondering.. all good if is not merged today

**Comment by streamer45** [comment]
> @streamer45 not the last chance no, I was just wondering.. all good if is not merged today

Alright then I'd say we can wait a little longer as the team is off these days and this PR should ideally go in with some other small changes that Chris mentioned (glare free negotiation). Thanks for the heads up though, really appreciated :)

**Comment by cpoile** [comment]
Thanks for the heads up @enahum

File: app/products/calls/connection/connection.ts
Comment (cpoile) [reply app/products/calls/connection/connection.ts:204]: I believe this was because we had to be absolutely sure the peer and tracks had been removed before calling stop, but it looks like RTCPeer simplifies this, so you're right -- no need for the callback. Removed.  
Working on the merge conflicts now.

**Comment by cpoile** [comment]
@larkox No, it's too late to add this big change, and we're waiting on another bug fix in a dependency before it makes sense to switch. We can wait for next release.

**Review by streamer45 — APPROVED** [review]
Looks good! 

@cpoile Do you know how much soaking time we'd get if we merged this asap? Looking to test if for a while before releasing.

**Comment by cpoile** [comment]
@streamer45 I'm not totally sure -- I asked @avinashlng1080 for the release schedule, he will know.

**Comment by avinashlng1080** [comment]
> @streamer45 I'm not totally sure -- I asked @avinashlng1080 for the release schedule, he will know.

hey @streamer45  and @cpoile  - if we merge this PR today, it will be bundle in the tomorrow's weekly build - the beta release

**Comment by cpoile** [comment]
@amyblais How much time will this have to soak with the beta users before it hits v2.1?

**Comment by enahum** [comment]
Code complete was yesterday.. any reason why it did not get merge?

**Comment by cpoile** [comment]
@enahum No particular reason, I think I misunderstood thinking it would be eod Jan 26.

**Comment by amyblais** [comment]
> @amyblais How much time will this have to soak with the beta users before it hits v2.1?

v2.1 is scheduled for Feb 16th. It would be good if this is included for tomorrow's build, otherwise I'd suggesting pushing this to v2.2.

**Comment by cpoile** [comment]
That's the goal -- we want it in beta first. So, I'll merge and I'll be quick to revert if something breaks.

**Comment by pbaveja** [comment]
Hey @cpoile, this upgrade to use the upstream version of react-native-webrtc seems to be breaking the android build here.
```
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':react-native-webrtc:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

I reverted the version back to the original mattermost fork and it built the APK successfully. 

Let me know how I can provide more details.

**Comment by cpoile** [comment]
@pbaveja I'm not sure what android build you're talking about, can you give more details please?

**Comment by streamer45** [comment]
@cpoile Could https://github.com/mattermost/mattermost-mobile/commit/64ee37dfd476a863a91b51abea47ad82ec7e995c be related? Our android CI build was failing as well at some point after this went in.

**Comment by cpoile** [comment]
@streamer45 Yep, that's what I suspected

**Comment by pbaveja** [comment]
@cpoile Sure.

I used the `main` branch of this repository. And after running the command `npm run build:android`, it failed with two errors.

- The first one being the error I mentioned previously:
`Execution failed for task ':react-native-webrtc:compileReleaseJavaWithJavac'`
- The second being a `Java StackOverflow (no error message`

Let me know what details from the build command output I can provide to help you

**Comment by cpoile** [comment]
@pbaveja It sounds like it might be https://github.com/mattermost/mattermost-mobile/commit/64ee37dfd476a863a91b51abea47ad82ec7e995c as Claudio mentioned -- we encountered the same issue. Hope that helps

**Comment by pbaveja** [comment]
@cpoile I ran the command mentioned here https://github.com/mattermost/mattermost-mobile/commit/64ee37dfd476a863a91b51abea47ad82ec7e995c

But the error persists :(

**Comment by streamer45** [comment]
@pbaveja Could you try to apply this change and see if it works? I had to do the same because the native libraries are not downloaded automatically.

```
diff --git a/scripts/build.sh b/scripts/build.sh
index a758a3854..127956fba 100755
--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -42,6 +42,7 @@ function setup() {
         npm run clean || exit 1
         npm install --ignore-scripts || exit 1
         npx patch-package || exit 1
+        node node_modules/react-native-webrtc/tools/downloadWebRTC.js || exit 1
 
         if [[ "$1" == "ios"* ]]; then
           echo "Installing Gems"
@@ -104,4 +105,4 @@ case $1 in
       apk: unsigned
       ipa: unsigned or simulator"
   ;;
-esac
\ No newline at end of file
+esac
```

**Comment by enahum** [comment]
> @pbaveja Could you try to apply this change and see if it works? I had to do the same because the native libraries are not downloaded automatically.
> 
> 
> 
> ```
> 
> diff --git a/scripts/build.sh b/scripts/build.sh
> 
> index a758a3854..127956fba 100755
> 
> --- a/scripts/build.sh
> 
> +++ b/scripts/build.sh
> 
> @@ -42,6 +42,7 @@ function setup() {
> 
>          npm run clean || exit 1
> 
>          npm install --ignore-scripts || exit 1
> 
>          npx patch-package || exit 1
> 
> +        node node_modules/react-native-webrtc/tools/downloadWebRTC.js || exit 1
> 
>  
> 
>          if [[ "$1" == "ios"* ]]; then
> 
>            echo "Installing Gems"
> 
> @@ -104,4 +105,4 @@ case $1 in
> 
>        apk: unsigned
> 
>        ipa: unsigned or simulator"
> 
>    ;;
> 
> -esac
> 
> \ No newline at end of file
> 
> +esac
> 
> ```

@streamer45 Can you please submit a PR with the patch? Thanks

**Pull request merged**
